packages/actions/src/contracts/governance.test.ts

```diff
@@ -16,42 +16,42 @@ const forkUrl = celo.rpcUrls.default.http[0]
 viem_testWithAnvil(
   'vote',
   async (client) => {
-    let walletClient: WalletCeloClient
+    let clients: { public: PublicCeloClient; wallet: WalletCeloClient }
     beforeEach(async () => {
-      client.impersonateAccount({ address: voter })
-      walletClient = createWalletClient({
+      await client.impersonateAccount({ address: voter })
+      const walletClient = createWalletClient({
         account: voter,
         chain: client.chain,
         transport: http(client.chain.rpcUrls.default.http[0]),
       })
+      clients = {
+        public: createPublicClient({
+          chain: client.chain,
+          transport: http(client.chain.rpcUrls.default.http[0]),
+        }),
+        wallet: walletClient,
+      }
     })
     afterEach(async () => {
       await client.stopImpersonatingAccount({ address: voter })
     })
     const txHashRegex = /^0x([A-Fa-f0-9]{64})$/
 
     it(
-      'votes Yes on a proposal',
+      'votes Yes/No on a proposal',
       async () => {
         // This will throw if proposalId is not in the dequeue
         // In a real test, ensure proposalId is valid and in the dequeue
-        await expect(vote(walletClient, proposalId, 'Yes')).resolves.toMatch(txHashRegex)
+        await expect(vote(clients, proposalId, 'Yes')).resolves.toMatch(txHashRegex)
+        await expect(vote(clients, proposalId, 'No')).resolves.toMatch(txHashRegex)
```

Comment:
i combined be cause A) little faster, and B) good to test that both can be called sequentially